Add first draft of learner-profiles

jansim opened the pull request:
Closes #664.


Based on issue #664, I'm adding some learner profiles for the course in place of the current (empty) placeholder.

This is my first time writing these sort of profiles, so I'd be happy for suggestions on improvements. I took these as inspirations:

- https://carpentries-incubator.github.io/python-packaging-publishing/learner-profiles/index.html
- https://github.com/carpentries-incubator/hpc-intro/blob/gh-pages/_extras/learner-profiles.md


<details>

For more guidance on how to contribute changes to a Carpentries project, please review [the Contributing Guide](CONTRIBUTING.md) and [Code of Conduct](https://docs.carpentries.org/topic_folders/policies/code-of-conduct.html).

Please keep in mind that lesson Maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum. If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact The Carpentries Team at team@carpentries.org.

</details>

vahtras commented:
Looks very good to me!

alee reviewed (commented):
I agree with @vahtras, this is excellent - thank you for taking the initiative to submit this PR @jansim ! 

The only suggestion I have would be to consider adding some language to the examples (perhaps just Maria's?) that makes clear that this lesson is a first step towards mastery, focused on developing a hopefully coherent mental model of plotting & programming in python so that additional learning can take place on the learner's own terms. Introducing primary concepts so that one can ask the right questions in future work, e.g., supporting future exploration. So they may still rely on colleagues for code support but they'll be able to ask better questions and know more about what may be possible in code to do with the data + analyses they are interested in.

Happy to take other community opinions here too!

jansim commented:
Thanks for the great feedback @alee ! I updated the profile to incorporate your feedback and fixed a naming inconsistency between the profile's title and body (renamed everything to Carla Correlation, as I couldn't come up with a good pun for Maria 😄).
